library/alloc/src/rc.rs

```diff
@@ -1075,7 +1075,7 @@ impl<T, A: Allocator> Rc<[T], A> {
                     Layout::array::<T>(len).unwrap(),
                     |layout| alloc.allocate_zeroed(layout),
                     |mem| {
-                        ptr::slice_from_raw_parts_mut(mem as *mut T, len)
+                        ptr::slice_from_raw_parts_mut(mem.cast::<T>(), len)
```

Comment:
Would adding a function call here reduce perf? Feels like a simple `as` would be better, but maybe since `cast` is just `as` wrapped in a function it'll get optimized away.

Reply:
There's no difference interchangeable, just nowadays the recommendation is to use `cast` since you can't can't accidentally change mutability that way